### ISSUE TITLE
`up -d --no-recreate` after removing container leads to `Invalid network mode: invalid container format container:<name|id>`

### ISSUE THREAD

**Issue by nocive** [docker/compose#9257]
**Description**

**Steps to reproduce the issue:**
1. Use the following compose project
```
version: '3.8'
services:
  foo:
    image: alpine
    command: tail -f /dev/null
    depends_on:
      memcached: {condition: service_healthy}
      redis: {condition: service_healthy}

  memcached:
    image: memcached:1.6-alpine
    stdin_open: true
    tty: true
    expose:
      - 11211
    healthcheck:
      test:     "echo | nc -w 5 localhost:11211"
      interval: 30s
      timeout:  5s
      retries:  5

  redis:
    image: redis:5.0.7-alpine
    stdin_open: true
    tty: true
    expose:
      - 6379
    healthcheck:
      test:     "echo | nc -w 5 localhost:6379"
      interval: 30s
      timeout:  5s
      retries:  5
```

2. `docker compose up -d --no-recreate foo`
3. `docker compose rm -fsv foo`
4. `docker compose up -d --no-recreate foo`

**Describe the results you received:**
```
[+] Running 4/4
 ⠿ Network dcompose-repro_default        Created
 ⠿ Container dcompose-repro-memcached-1  Healthy
 ⠿ Container dcompose-repro-redis-1      Healthy
 ⠿ Container dcompose-repro-foo-1        Started
[+] Running 1/1
 ⠿ Container dcompose-repro-foo-1  Stopped
Going to remove dcompose-repro-foo-1
[+] Running 1/0
 ⠿ Container dcompose-repro-foo-1  Removed
[+] Running 0/0
 ⠋ Container dcompose-repro-foo-1  Creating
Error response from daemon: Invalid network mode: invalid container format container:<name|id>
```

**Describe the results you expected:**
A successfully up project

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker compose version`:**

```
Docker Compose version v2.3.3 (but also occurs in previous versions)
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., v0.6.1-docker)
  compose: Docker Compose (Docker Inc., v2.3.3)

Server:
 Containers: 82
  Running: 1
  Paused: 0
  Stopped: 81
 Images: 182
 Server Version: 20.10.11
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: false
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2 io.containerd.runtime.v1.linux
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 1e5ef943eb76627a6d3b6de8cd1ef6537f393a71.m
 runc version: v1.0.3-0-gf46b6ba2
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.15.7-arch1-1
 Operating System: Antergos Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.49GiB
 Name: xxx
 ID: xxx
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: nocive
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by olemoign** [comment]
Probably linked to #8940.

**Comment by nocive** [comment]
Seems to be fixed in 2.4.0, can't reproduce it anymore. Closing.
